### PR TITLE
Add makefile to CI\CD

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

Add makefile to allow dynamic creating of Review appliaton. The Make file aslo makes it easily to call Terraform commands from the commandLine without supplying lengthy terraform arguments. 

### Changes proposed in this pull request

add makefile

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,13 +33,15 @@ yarn-debug.log*
 *.log
 .envrc
 terraform/app_secrets.yml
+*token*
```
Comment (gpeng) [inline .gitignore:36]: This is too heavy handed. We have `app/services/dttp/access_token_service.rb` which will be ignored. We can't ignore everything with `token` in it.

File: .gitignore [hunk 2]
```diff
@@ -33,13 +33,15 @@ yarn-debug.log*
...
 
 #terraform files
 .terraform/
 tfplan
 *.tfplan
 *.plan
-*secrets.tfvars
+*secrets*
```
Comment (gpeng) [inline .gitignore:43]: Same here. We shouldn't ignore everything that has `secrets` in the file name.

**Comment by felixtheflex** [comment]
[Add MakeFile for creating Isolated environment](https://trello.com/c/O30LeFXG/604-add-makefile-for-creating-isolated-environment)

File: .gitignore (hunk 1 above)
Comment (felixtheflex) [reply .gitignore:36]: cool... noted.

Comment (felixtheflex) [reply .gitignore:36]: now relexed..... in hindsight, it was heavy handed. i guess i was too cautious

File: .gitignore (hunk 2 above)
Comment (felixtheflex) [reply .gitignore]: resolved...

File: terraform/workspace-variables/production.tfvars [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # PaaS
-paas_sso_passcode                = ""
+#paas_sso_passcode                = ""
```
Comment (saliceti) [inline terraform/workspace-variables/production.tfvars:2]: Remove comment

File: terraform/workspace-variables/review.tfvars [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# PaaS
+#paas_sso_passcode                = ""
```
Comment (saliceti) [inline terraform/workspace-variables/review.tfvars:2]: Remove comment

File: terraform/workspace-variables/review.tfvars [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#paas_app_environment             = "qa"
```
Comment (saliceti) [inline terraform/workspace-variables/review.tfvars:3]: Can you replace comment with something along the lines of
```
#paas_app_environment not specified here, must be passed to terraform dymamically
```

File: terraform/workspace-variables/review.tfvars [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+paas_app_start_timeout           = "60"
+paas_web_app_deployment_strategy = "blue-green-v2"
+paas_web_app_instances           = 1
+paas_web_app_memory              = 512
+paas_worker_app_instances        = 10
+paas_worker_app_memory           = 512
+
+statuscake_alerts = {
+
+  #  alert = {
```
Comment (saliceti) [inline terraform/workspace-variables/review.tfvars:16]: Remove comment

File: terraform/workspace-variables/review_backend.tfvars [hunk 7]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "s121d01-register"
+storage_account_name = "s121d01registerterraform"
+container_name       = "tfstate"
+#key                  = "qa-bat.terraform.tfstate"
```
Comment (saliceti) [inline terraform/workspace-variables/review_backend.tfvars:4]: Can you replace comment with something along the lines of
```
#key not specified here, must be passed to terraform dymamically
```

File: terraform/workspace-variables/staging.tfvars [hunk 8]
```diff
@@ -1,6 +1,7 @@
 # PaaS
-paas_sso_passcode                = ""
+#paas_sso_passcode                = ""
```
Comment (saliceti) [inline terraform/workspace-variables/staging.tfvars:2]: Remove comment

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,34 @@
+review:
+	echo "setting Review $(env) environment"
+	$(eval env_config=review)
+	$(eval backend_key=-backend-config=key=$(env).terraform.tfstate)
+	$(eval export TF_VAR_paas_app_environment=$(env))
+
+qa:
+	$(eval env=qa)
+	$(eval env_config=qa)
```
Comment (saliceti) [inline Makefile:9]: `env_config` was added to the tfvars, so there is no need to ad it here. `$(eval export TF_VAR_env_config=$(env_config))` can be moved to `review:`

File: terraform/workspace-variables/production.tfvars (hunk 3 above)
Comment (felixtheflex) [reply terraform/workspace-variables/production.tfvars]: done...

File: terraform/workspace-variables/review.tfvars (hunk 5 above)
Comment (felixtheflex) [reply terraform/workspace-variables/review.tfvars]: done

File: terraform/workspace-variables/review.tfvars (hunk 6 above)
Comment (felixtheflex) [reply terraform/workspace-variables/review.tfvars]: done....

File: terraform/workspace-variables/review_backend.tfvars (hunk 7 above)
Comment (felixtheflex) [reply terraform/workspace-variables/review_backend.tfvars]: done ...

File: Makefile (hunk 9 above)
Comment (felixtheflex) [reply Makefile:12]: done please check....

Comment (saliceti) [reply Makefile:12]: well it's still here: `$(eval env_config=qa)`

Comment (saliceti) [reply Makefile:12]: and on line 39

File: terraform/workspace-variables/qa.tfvars [hunk 10]
```diff
@@ -1,6 +1,7 @@
 # PaaS
 paas_sso_passcode                = ""
```
Comment (saliceti) [inline terraform/workspace-variables/qa.tfvars:2]: This shouldn't be here

File: terraform/workspace-variables/review.tfvars [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+paas_postgres_service_plan       = "tiny-unencrypted-11"
+paas_redis_service_plan          = "tiny-4_x"
+paas_app_start_timeout           = "60"
+paas_web_app_deployment_strategy = "blue-green-v2"
+paas_web_app_instances           = 1
+paas_web_app_memory              = 512
+paas_worker_app_instances        = 10
+paas_worker_app_memory           = 512
+
+statuscake_alerts = {
```
Comment (saliceti) [inline terraform/workspace-variables/review.tfvars:13]: You can remove make it prettier `statuscake_alerts = {}`

File: Makefile (hunk 9 above)
Comment (felixtheflex) [reply Makefile:10]: done.

File: terraform/workspace-variables/review.tfvars (hunk 11 above)
Comment (felixtheflex) [reply terraform/workspace-variables/review.tfvars]: done

**Review by saliceti — APPROVED** [review]

**Pull request merged**